### DISCUSSION TITLE
Use XIRR for performance calculation

### DISCUSSION THREAD

**Discussion by thelazyoxymoron** [ghostfolio/ghostfolio#2516] [Ideas]
I noticed that all the calculations were done with absolute percentage returns. As a value investor, I feel focusing on XIRR would give a more accurate insight into the portfolio return.

**Comment by fmnamado** [comment]
Hello to you all,

First of all, thank you for your work.

I will take this thread to post my idea, as I agree with the OP, that XIRR should be used, at least it should be displayed.
It doesn't have to replace the other calculations or values, but it should be another one to be displayed.

The interest of having this solution is to see the XIRR so I can follow my portfolio annualized rates, updated in terms of time and cash flows.
Some info here: https://marcellus.helpscoutdocs.com/article/65-twrr-xirr

Now I will give you some example, a practical one, so I can illustrate the need of XIRR.

Today is the 22nd of January 2024, and IWDA.AS has the market price of 83,89EUR
To keep this simple, I will only buy and sell just one unit at a time.

Let's pretend I traded this in the past, always with 10% annualized gains.

Of the value is negative, consider it a BUY, a positive is SELL or the current portfolio value.
The values should be this
```
22/01/2020	-57,29
22/01/2021	63,02
22/01/2022	-69,33
22/01/2023	-76,26
22/01/2024	167,78
```

This means in english:
```
22nd of January 2020 - BUY 1 @ 57.29 EUR
22nd of January 2021 - SELL 1 @ 63.02 EUR (locked a 10% gain and closed the position)
22nd of January 2022 - BUY 1 @ 69.33 EUR
22nd of January 2023 - BUY 1 @ 76.26 EUR (10% increase value of the previous year)
22nd of January 2020 - SELL 2 @ 2*83.89 EUR (10% increase value of the previous year and closed the position)
```
the last trade is not a real one, it is @ today's value, but I have to consider this in excel, the current value as it it would be a trade.

If I do a XIRR on excel with this values, I get a 9,9948% XIRR gain, which is what I expect.
The first two trades would close a position with 10% gain: either it is considered to the total XIRR or it would be shown as realized gains (???)
It it isn't considered, for me it is the same as to just consider just the open positions, because the first on is closed and "archived":

```
22nd of January 2022 - BUY 1 @ 69.33 EUR
22nd of January 2023 - BUY 1 @ 76.26 EUR (10% increase value of the previous year)
22nd of January 2020 - SELL 2 @ 2*83.89 EUR (10% increase value of the previous year and closed the position)
```
10% gain times a 10% gain what would be a 10% annualized and XIRR gain.


So If I insert just this data down below in ghost folio:
```
22nd of January 2020 - BUY 1 @ 57.29 EUR
22nd of January 2021 - SELL 1 @ 63.02 EUR (locked a 10% gain and closed the position)
22nd of January 2022 - BUY 1 @ 69.33 EUR
22nd of January 2023 - BUY 1 @ 76.26 EUR (10% increase value of the previous year)
```
I get:
```
Time in Market about 4 years
4 transactions
Buy 202.71 EUR
Sell 63.02 EUR
Investment 139.69 EUR
Absolute Gross Performance 28.06 EUR
Gross Performance (TWR) + 30.97%
Fees - 0.00 EUR
Absolute Net Performance 28.06 EUR
Net Performance (TWR) + 30.97%
Total Assets 167.75 EUR
Valuables 0.00 EUR
Emergency Fund 0.00 EUR
Cash 0.00 EUR
Assets 0.00 EUR
Buying Power 0.00 EUR
Excluded from Analysis0.00 EUR
Liabilities -0.00 EUR
Net Worth 167.75 EUR
Annualized Performance +6.97%
Interest 0.00EUR
Dividend 0.00EUR 
```
I would expect to see a 10% annualized gain, either be it in Annualized Performance or some XIRR calculated field.

I am not sure if I am being clear, but would love to hear your feedback.
Thank you!
